tests/test_iis.py

```diff
@@ -1,58 +1,153 @@
+    'MoveRequestsPersec': 0,
+    'Name': "Default Web Site",
+    'NonAnonymousUsersPersec': 0,
+    'NotFoundErrorsPersec': 0,
+    'OptionsRequestsPersec': 0,
+    'OtherRequestMethodsPersec': 0,
+    'PostRequestsPersec': 0,
+    'PropfindRequestsPersec': 0,
+    'ProppatchRequestsPersec': 0,
+    'PutRequestsPersec': 0,
+    'SearchRequestsPersec': 0,
+    'ServiceUptime': 251,
+    'TotalAllowedAsyncIORequests': 0,
+    'TotalAnonymousUsers': 0,
+    'TotalBlockedAsyncIORequests': 0,
+    'Totalblockedbandwidthbytes': 0,
+    'TotalBytesReceived': "0",
+    'TotalBytesSent': "0",
+    'TotalBytesTransferred': "0",
+    'TotalCGIRequests': 0,
+    'TotalConnectionAttemptsallinstances': 0,
+    'TotalCopyRequests': 0,
+    'TotalcountoffailedCALrequestsforauthenticatedusers': 0,
+    'TotalcountoffailedCALrequestsforSSLconnections': 0,
+    'TotalDeleteRequests': 0,
+    'TotalFilesReceived': 0,
+    'TotalFilesSent': 0,
+    'TotalFilesTransferred': 0,
+    'TotalGetRequests': 0,
+    'TotalHeadRequests': 0,
+    'TotalISAPIExtensionRequests': 0,
+    'TotalLockedErrors': 0,
+    'TotalLockRequests': 0,
+    'TotalLogonAttempts': 0,
+    'TotalMethodRequests': 0,
+    'TotalMethodRequestsPersec': 0,
+    'TotalMkcolRequests': 0,
+    'TotalMoveRequests': 0,
+    'TotalNonAnonymousUsers': 0,
+    'TotalNotFoundErrors': 0,
+    'TotalOptionsRequests': 0,
+    'TotalOtherRequestMethods': 0,
+    'TotalPostRequests': 0,
+    'TotalPropfindRequests': 0,
+    'TotalProppatchRequests': 0,
+    'TotalPutRequests': 0,
+    'TotalRejectedAsyncIORequests': 0,
+    'TotalSearchRequests': 0,
+    'TotalTraceRequests': 0,
+    'TotalUnlockRequests': 0,
+    'TraceRequestsPersec': 0,
+    'UnlockRequestsPersec': 0,
+}
+
+
+class Mocked_Win32_PerfFormattedData_W3SVC_WebService(object):
+    """
+    Generate Mocked instance of Win32_PerfFormattedData_W3SVC_WebService
+    """
+    def __init__(self, **entries):
+        self.__dict__.update(entries)
+
+
+class Mocked_WMI(Mock):
+    """
+    Mock WMI methods for test purpose
+    """
+    def Win32_PerfFormattedData_W3SVC_WebService(self):
+        """
+        Returns mock match Win32_PerfFormattedData_W3SVC_WebService
+        """
+        return [Mocked_Win32_PerfFormattedData_W3SVC_WebService(
+                **Win32_PerfFormattedData_W3SVC_WebService_attr)]
+
+
+class IISSTestCase(AgentCheckTest):
+    CHECK_NAME = 'iis'
+
+    WIN_SERVICES_CONFIG = {
+        'host': ".",
+        'tags': ["mytag1", "mytag2"],
+        'sites': ["Default Web Site", "Failing site"]
+    }
+
+    def test_check(self):
+        """
+        Returns the right metrics and service checks
+        """
+        # Mocking `wmi` Python package
+        import sys
+        sys.modules['wmi'] = Mocked_WMI()
+
+        # Run check
+        config = {
+            'instances': [self.WIN_SERVICES_CONFIG]
+        }
+
+        self.run_check_twice(config)
+
+        # Test metrics
+        for mname, _, _ in self.check.METRICS:
```

Comment:
You should probably put a copy of this list in the test.
Otherwise one may remove a metric accidentally and the test would still pass.